Source/core/maps/GoogleMapsManager.js

```diff
@@ -99,32 +151,80 @@ GoogleMapsManager.prototype.convertDistance = function(distanceStr) {
 };
 
 /**
+ * Converts total time into hours.
+ * @param {string} timeStr
+ * @return {float} hrs
+ */
+
+GoogleMapsManager.prototype.convertTime = function(timeStr) {
+  if (timeStr) {
+    var days = (/ (\w*) d/).exec(timeStr);
+    var hrs = (/ (\w*) h/).exec(timeStr);
+    var mins = (/ (\w*) m/).exec(timeStr);
+    if(hrs){
+      hrs = parseFloat(hrs[1]);
+    }
+    else{
+      hrs = 0;
+    }
+    if(mins){
+      mins = parseFloat(mins[1]);
+      hrs += mins/60;
+    }
+    if(days){
+      days = parseFloat(days[1]);
+      hrs += days*24;
+    }
+    return hrs;
+  }
+};
+
+/**
  * Inserts element where footprints will be displayed if not present
  * @param {object} route
  * @param {element} e
+ * @param {string} type
  */
 
-GoogleMapsManager.prototype.insertFootprintElement = function(route, e) {
+GoogleMapsManager.prototype.insertFootprintElement = function(route, e, type) {
```

Comment:
Note how you have to pass "type" as an extra argument to every function in the program... And in many cases, such as here, it is not conceptually clear why you have to do it.

There must be a better way to do this...